### DISCUSSION TITLE
Laravel Breeze api with Next Auth

### DISCUSSION THREAD

**Discussion by samushi** [nextauthjs/next-auth#4900] [Help]
### Question 💬

My question is short, how can I integrate Laravel breeze with next-auth, in this case, the sessions open on the server-side
https://laravel.com/docs/9.x/starter-kits#breeze-and-next

### How to reproduce ☕️

https://laravel.com/docs/9.x/starter-kits#breeze-and-next

### Contributing 🙌🏽

Yes, I am willing to help answer this question in a PR

**Comment by ndom91** [comment]
So I'm not very familiar with laravel or breeze especially, but it seems that breeze is based on phps default cookie based session handling, right? 

It seems they also have a oauth provider type plugin though. See: https://laravel.com/docs/9.x/passport 

It explicitly says breeze and passport oauth based auth are NOT mutually exclusive (https://laravel.com/docs/9.x/authentication#laravels-api-authentication-services)

If you can get breeze+passport setup as a custom oauth provider, then you can use next-auths custom oauth provider (https://next-auth.js.org/configuration/providers/oauth#using-a-custom-provider) and plugin the oauth URLs provided by breeze+passport probably. 

Again, I'm not super familiar, but digging through the docs for 5-10min looks like this _could_ be possible 👍 

**EDIT**: Alternatively you could use next-auth's credentials provider where you basically control the sending of credentials back n forth and returning `true`/`false` whether the user is authenticated or not (https://next-auth.js.org/configuration/providers/credentials). And then you would communicate with your breeze api routes in the `authorize` method of the credentials provider, as shown in the example in the link, and return `true`/`false` from the authorize method depending on the breeze api response(s).

It would probably look something like their example apps `withAuth` hook and `login` method. 

Hook: https://github.com/laravel/breeze-next/blob/master/src/hooks/auth.js#L37
Usage: https://github.com/laravel/breeze-next/blob/master/src/pages/login.js#L38

**Comment by samushi** [comment]
This is the way I am trying to integrate, this is how it works, but the problem here is that the sessions are not accessing axios in the component.

```
export default async function auth(
  request: NextApiRequest,
  response: NextApiResponse
) {
  return await NextAuth(request, response, {
    providers: [
      CredentialProvider({
        id: "login",
        name: "login",
        credentials: {
          email: {
            label: "Email",
            type: "email",
            placeholder: "Your email address",
          },
          password: {
            label: "Password",
            type: "password",
            placeholder: "Your password",
          },
        },
        authorize: async (credentials: any, req: any) => {
          try {

            const cookies = await axios.get("/sanctum/csrf-cookie")
            const xsrf = cookies.headers["set-cookie"] as string[]
            const token = decodeURIComponent(
              xsrf[0].split(";")[0].split("=")[1]
            )

            const login = await axios.post("/login", credentials, {
              headers: {
                "Content-Type": "application/json",
                Cookie: `${decodeURIComponent(xsrf[0])}; ${decodeURIComponent(
                  xsrf[1]
                )}`,
                "X-XSRF-TOKEN": token,
              },
            })
            const loginCookies = login.headers["set-cookie"] as string[]
            const xsrfToken = decodeURIComponent(loginCookies[0])
            const sessionToken = decodeURIComponent(loginCookies[1])

            if (loginCookies.length > 0 && response) {
              // Set cookies
              response.setHeader("Set-Cookie", [
                `${decodeURIComponent(xsrfToken)}`,
                `${decodeURIComponent(sessionToken)}`,
              ])

              const user = await axios.get("/api/user", {
                headers: {
                  Cookie: `${xsrfToken}; ${sessionToken}`,
                  Referer: "http://localhost:3000/",
                },
              })

              const data = {
                user: user.data,
              }
              return data
            }
          } catch (error) {
            return null
          }

          return null
        },
      }),
    ],
    secret: process.env.NEXTAUTH_SECRET,
    pages: {
      signIn: "login",
      error: "login",
    },
  })
```

**Reply by the-programmer-alex** [reply]
See [this stackoverflow](https://stackoverflow.com/questions/69958294/next-auth-credentials-provider-with-laravel-request-failed-with-status-code-41)., might help

**Comment by DenisAlliswell** [comment]
Hey, @samushi ! Any success on your integration ? If so, could you please somehow share your working code?

**Comment by d8vjork** [comment]
After months trying everything, I surrender, just remove this package as nothing works

**What I expect:** Anything related to OAuth 2.0 to work within, what I get? Nothing works.... Tried even debugging NodeJS but no luck, apparently something internal to this package

No idea why is so difficult to use this as a simple client, guess is not that simple...
